src/components/main/files/upload/mod.rs

```diff
@@ -100,3 +114,39 @@ pub fn Upload<'a>(cx: Scope<'a, Props<'a>>) -> Element<'a> {
         ))
     })
 }
+
+
+async fn update_thumbnail(file_storage: Storage, filename_to_save: String) -> Result<String, Error> {
+    let item =  file_storage.root_directory().get_item(&filename_to_save)?;
+    let parts_of_filename: Vec<&str> = filename_to_save.split('.').collect();
+
+    //Since files selected are filtered to be jpg, jpeg, png or svg the last branch is not reachable
+    let mime = match parts_of_filename.last() {
```

Comment:
Should use `to_lowercase()` as a map in the event the extension of the file is uppercase or a mix case.